### PR TITLE
Rubicon Bid Adapter: pass DSA fields

### PR DESCRIPTION
**Pull request by apukh-magnite**
## Type of change
- [X] Feature

## Description of change
The rubicon prebid.js adapter must be able to pass publisher-specified DSA fields through the fastlane.json GET interface.

### CODE REVIEW

**Review by patmmccann — APPROVED** [review]

**Comment by patmmccann** [comment]
Are you handling the response in a different pr?

**Comment by robertrmartinez** [comment]
We had a ticket to pass through I do not recall anything about response!

But makes sense haha. Will connect with team and get back.

Thanks @patmmccann

**Comment by patmmccann** [comment]
https://github.com/prebid/Prebid.js/pull/10971/files#diff-a2981000898eb78cc9ada7db90cde0b6eb4fd7e9497d121312609651042c7980R619

File: test/spec/modules/rubiconBidAdapter_spec.js [hunk 1]
```diff
@@ -1574,7 +1575,7 @@ describe('the rubicon adapter', function () {
           describe('UserID catchall support', function () {
             it('should send user id with generic format', function () {
               const clonedBid = utils.deepClone(bidderRequest.bids[0]);
-              // Hardcoding userIdAsEids since createEidsArray returns empty array if source not found in eids.js
+              // Hardcoding userIdAsEids since createEirray returns empty array if source not found in eids.js
```
Comment (robertrmartinez) [inline test/spec/modules/rubiconBidAdapter_spec.js:1578]: can you revert this back to correct spelling

File: test/spec/modules/rubiconBidAdapter_spec.js [hunk 2]
```diff
@@ -1590,7 +1591,7 @@ describe('the rubicon adapter', function () {
 
             it('should send rubiconproject special case', function () {
               const clonedBid = utils.deepClone(bidderRequest.bids[0]);
-              // Hardcoding userIdAsEids since createEidsArray returns empty array if source not found in eids.js
+              // Hardcoding userIdAsEids since createEirray returns empty array if source not found in eids.js
```
Comment (robertrmartinez) [inline test/spec/modules/rubiconBidAdapter_spec.js:1594]: can you revert this back to correct spelling `createEidsArray` is correct

File: test/spec/modules/rubiconBidAdapter_spec.js [hunk 3]
```diff
@@ -3286,6 +3338,91 @@ describe('the rubicon adapter', function () {
...
+                ],
+                'dsa': {}
+              }
+            ]
+          };
+          // config.setConfig({
+          //   rubicon: {
+          //     netRevenue: false
+          //   }
+          // });
```
Comment (robertrmartinez) [inline test/spec/modules/rubiconBidAdapter_spec.js:3414]: remove this commented out stuff

File: test/spec/modules/rubiconBidAdapter_spec.js [hunk 4]
```diff
@@ -3286,6 +3338,91 @@ describe('the rubicon adapter', function () {
...
+          let bids = spec.interpretResponse({body: response}, {
+            bidRequest: bidderRequest.bids[0]
+          });
+          expect(bids).to.be.lengthOf(2);
+          expect(bids[1].meta.dsa).to.have.property('behalf');
+          expect(bids[1].meta.dsa).to.have.property('paid');
+
+          // if we dont have dsa field in response or the dsa object is empty
+          expect(bids[0].meta).to.not.have.property('dsa');
```
Comment (robertrmartinez) [inline test/spec/modules/rubiconBidAdapter_spec.js:3423]: seems strange that the input ads[0] has DSA and ads[1] does not

But bids[0] does not have DSA but ads[1] does.

But I guess it has something to do with the input `bidRequest` into the `spec.interpretResponse` above?

Anyways, seems like it works and tests pass

Comment (apukh-magnite) [reply test/spec/modules/rubiconBidAdapter_spec.js:3418]: The comments have been updated.

**Review by robertrmartinez — APPROVED** [review]
looks good to me!

**Comment by robertrmartinez** [comment]
Needs non-magnite 2nd reviewer!

**Comment by robertrmartinez** [comment]
@ChrisHuie @patmmccann 

Could one of you give this a second non magnite review please! Or we can assign to someone else.

Want to get this into this weeks release (I am on duty so will do release once this is in) Might wait until Thursday maybe.

**Review by ChrisHuie — APPROVED** [review]

**Pull request merged**